controllers/preCachingConfig.go

```diff
@@ -0,0 +1,80 @@
+package controllers
+
+import (
+	"context"
+	"reflect"
+
+	ranv1alpha1 "github.com/openshift-kni/cluster-group-upgrades-operator/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+// getPreCachingConfig: retrieves pre-caching configuration CR
+func (r *ClusterGroupUpgradeReconciler) getPreCachingConfig(
+	ctx context.Context, clusterGroupUpgrade *ranv1alpha1.ClusterGroupUpgrade) (
+	*ranv1alpha1.PreCachingConfig, error) {
+
+	preCachingConfig := new(ranv1alpha1.PreCachingConfig)
+	preCachingConfigRef := clusterGroupUpgrade.Spec.PreCachingConfigRef
+
+	preCachingConfigName := preCachingConfigRef.Name
+	preCachingConfigNamespace := preCachingConfigRef.Namespace
+	// If namespace is not specified, assume the CGU namespace
+	if preCachingConfigNamespace == "" {
+		preCachingConfigNamespace = clusterGroupUpgrade.Namespace
+	}
+
+	if preCachingConfigName == "" {
+		r.Log.Info("getPreCachingConfigSpec: no preCachingConfig CR specified")
+		return preCachingConfig, nil
+	}
+
+	err := r.Get(ctx, types.NamespacedName{Namespace: preCachingConfigNamespace,
+		Name: preCachingConfigName}, preCachingConfig)
+
+	if err != nil {
+		if errors.IsNotFound(err) {
+			return preCachingConfig, nil
```

Comment:
Shouldn't this be treated as an error?

Reply:
My initial implementation did treat the `IsNotFound` as an error, however, I then decided to follow the same process as that for retrieving the cluster-group-upgrade-overrides ConfigMap, ref: https://github.com/openshift-kni/cluster-group-upgrades-operator/blob/main/controllers/overrides.go#L36-L40. 

I can change this back to return the error instead. Please confirm.

Reply:
Config map wasn't a feature published in the API. Its name is fixed and it's expected that it doesn't always exist. It's totally different from the new config CR that is specified by customer in the actual CGU spec.

Reply:
Very good point! In that case the not found resource should be reported as an error! I will change this.